File: js/euler03.js
function euler03(){
    var sq = Math.round(Math.sqrt(600851475143));
    var sieved = new Array(sq + 1).fill(false);
    var res = 0;
    for (let i = 2; i < sq; i++){
        if (!sieved[i]){
            for (let j = 2 * i; j <= sq; j += i){
                sieved[j] = true;
            }
            if (600851475143 % i == 0){
                res = i;
            }
        }
    }
    return res;
}

console.log(euler03());

File: go/euler03.go
package main
import (
    "fmt"
    "math"
)

func main() {
    res := 0
    n := 600851475143
    s := int(math.Floor(math.Sqrt(float64(n))))
    sieved := make([]bool, s)
    for i := 2; i < len(sieved); i++ {
        if (!sieved[i]){
            for j := 2 * i; j < len(sieved); j += i {
                sieved[j] = true
            }
            if n % i == 0 {
                res = i
            }
        }
    } 
    fmt.Println(res)
}
